plugins/modules/ec2_vpc_vgw.py

```diff
@@ -198,39 +212,50 @@ def get_vgw_info(vgws):
         return vgw_info
 
 
-def wait_for_status(client, module, vpn_gateway_id, status):
+def wait_for_status(
+    client, module: AnsibleAWSModule, vpn_gateway_id: Union[str, List[str]], desired_status: str
+) -> Tuple[bool, Any]:
     polling_increment_secs = 15
     max_retries = module.params.get("wait_timeout") // polling_increment_secs
-    status_achieved = False
+    try:
+        wait_for_resource_state(client, module, "vpn_gateway_exists", VpnGatewayIds=vpn_gateway_id)
```

Comment:
```suggestion
        wait_for_resource_state(client, module, "vpn_gateway_exists", VpnGatewayIds=[vpn_gateway_id])
```

Reply:
not sure what happens with the API but it fails with
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter VpnGatewayIds[0], value: ['vgw-0849b49ec813d18ee'], type: <class 'list'>, valid types: <class 'str'>
```

Reply:
According to the API, VpnGatewayIds must be a list of strings.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/describe_vpn_gateways.html
The error you get may be because you are not passing a list of strings. You need to verify that a list of strings is always being passed.

Reply:
Hi @alinabuzachis , fixed. It was indeed being passed as List[Str] at the API call.
Only thing was that it was handled to be passed as List[Str] earlier in the function call tree.
Updated the code to ensure it is passed as List[Str] before calling module_utils/ec2 functions.
https://github.com/ansible-collections/community.aws/pull/2171/commits/27b07e056bccfd390b8c1425acfdaa6ac58a44af